src/main/kotlin/org/bundleproject/plugins/Routing.kt

```diff
@@ -1,15 +1,99 @@
     routing {
         get("/") {
-            call.respondText("Hello World!")
+            TODO("Send docs here")
+        }
+        get("/v1/mods/{id}/{platform}/{minecraftVersion}/{version}") {
+            val assets = fetchAssets()
+            val id = call.parameters["id"]!!
+            val platform = call.parameters["platform"]!!
+            val minecraftVersion = call.parameters["minecraftVersion"]!!
+            val version = call.parameters["version"]!!
+            val mod = assets.mods[id]
+            val modData = mod
+                ?.platforms
+                ?.get(platform)
+                ?.get(minecraftVersion)
+                ?.get(version)
+                ?: return@get call.respond(
+                    HttpStatusCode.NotFound,
+                    ErrorResponse(
+                        error = "Invalid mod"
+                    )
+                )
+            try {
+                call.respond(
+                    HttpStatusCode.OK,
+                    ModResponse(
+                        data = ModResponseData(
+                            url = resolveUrl(ModData(
+                                version = version,
+                                source = modData.source,
+                                ref = modData.ref,
+                                name = id,
+                                id = modData.id
+                            )),
+                            metadata = mod.metadata
+                        )
+                    )
+                )
+            } catch (ignored: ModNotFoundException) {
+                call.respond(
+                    HttpStatusCode.InternalServerError,
+                    ErrorResponse(
+                        "Failed to fetch data url from source"
+                    )
+                )
+            }
+        }
+        get("/v1/mods/{id}/{platform}/{minecraftVersion}/{version}/download") {
+            val assets = fetchAssets()
+            val id = call.parameters["id"]!!
+            val platform = call.parameters["platform"]!!
+            val minecraftVersion = call.parameters["minecraftVersion"]!!
+            val version = call.parameters["version"]!!
+            val mod = assets.mods[id]
+            val modData = mod
+                ?.platforms
+                ?.get(platform)
+                ?.get(minecraftVersion)
+                ?.get(version)
+                ?: return@get call.respond(
+                    HttpStatusCode.NotFound,
+                    ErrorResponse(
+                        error = "Invalid mod"
+                    )
+                )
+            try {
+                call.respondRedirect(resolveUrl(ModData(
+                    version = version,
+                    source = modData.source,
+                    ref = modData.ref,
+                    name = id,
+                    id = modData.id
+                )), permanent = true)
+            } catch (ignored: ModNotFoundException) {
```

Comment:
Instead of using a try-catch could we utilise https://ktor.io/docs/status-pages.html